src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Color = 'red' | 'green' | 'blue';
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+
 export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
+}
 
+function isBadTriangle(
+  side1: number,
+  side2: number,
+  side3: number,
+): boolean {
+  return (side1 + side2) <= side3
+    || (side1 + side3) <= side2
+    || (side2 + side3) <= side1;
```

Comment:
You can check it by getting largest side with `Math.max(a, b, c)` and calculating only once:
`(a + b + c - largest) <= largest`